### PR TITLE
Add Decider impl for Nova onchain

### PR DESCRIPTION
**Pull request by arnaucube**
- Add Decider impl for Nova onchain's circuit.
- Update also the Decider trait.

Nova onchain decider: (compressed SNARK / final proof), in order to later verify the Nova+CycleFold proofs onchain (in Ethereum’s EVM).

As a sidenote info, these past days I've been benchmarking the decider circuit, and next thing I'll do is to work on the step 6 of the decider circuit ([src/folding/nova/decider_eth_circuit.rs#L418](https://github.com/privacy-scaling-explorations/folding-schemes/blob/383a70de2f2d72ed0fa70d8423bc27fb12898a66/src/folding/nova/decider_eth_circuit.rs#L418)) because the current approach takes too many constraints.

### CODE REVIEW

File: src/folding/circuits/nonnative.rs [hunk 1]
```diff
@@ -54,12 +54,26 @@ where
...
 ) -> Result<(Vec<C::ScalarField>, Vec<C::ScalarField>), SynthesisError>
+where
+    <C as ark_ec::CurveGroup>::BaseField: ark_ff::PrimeField,
+{
+    point_to_nonnative_limbs_custom_opt(p, OptimizationType::Weight)
+}
+
+/// point_to_nonnative_limbs_custom_opt is used to compute (outside the circuit) the limbs
+/// representation of a point that matches the one used in-circuit, and in particular this method
+/// allows to specify which OptimizationType to use.
```
[the remark this answers is not in the document]
Comment (arnaucube) [reply src/folding/circuits/nonnative.rs:71]: Nice!

File: Cargo.toml [hunk 2]
```diff
@@ -26,17 +27,17 @@ espresso_subroutines = {git="https://github.com/EspressoSystems/hyperplonk", pac
...
 
 [features]
 default = ["parallel"]
 
-parallel = [ 
-    "ark-std/parallel", 
-    "ark-ff/parallel",  
-    "ark-poly/parallel", 
-    ]
+parallel = []
```
[the remark this answers is not in the document]
Comment (arnaucube) [reply Cargo.toml]: Right, I did some late tests and forgot to re-add them, thanks for pointing it out. Also added `"ark-ec/parallel", "ark-crypto-primitives/parallel", "ark-r1cs-std/parallel"` to the list.

**Comment by arnaucube** [comment]
> Unsure why do we add the MNT curves here. To avoid Foreign Field ops in-circuit? If not can you explain?

Ideally I would have avoided the use of the MNT curves, and would have used directly BN254 - Grumpkin cycle, but at the latest release of arkworks (v0.4.0) the BN254 does not have the constraints implemented, which was added recently (by you [here](https://github.com/arkworks-rs/algebra/pull/737) hehe), so once we migrate to the new arkworks version (which is not yet released) we can enjoy the BN254 R1CS constraints. But for the moment for this test needed a cycle of curves with pairings, so used the MNT cycle for the test.

Regarding arkworks versions, there have been substantial improvements and fixes on many arkworks crates over this past year (since v0.4.0) release, just sent a msg in arkworks chat asking if there are plans to do a new release with all those changes that we would benefit from (eg. the mentioned bn254 constraints, some fixes, and maybe also the new configurable nonnativefield gadgets).

**Review by han0110 — APPROVED** [review]
LGTM!

**Pull request merged**